spec/jobs/socure_shadow_mode_proofing_job_spec.rb

```diff
@@ -0,0 +1,384 @@
+              success: true,
+              errors: {},
+              exception: nil,
+              timed_out: false,
+              transaction_id: 'resolution-mock-transaction-id-123',
+              reference: 'aaa-bbb-ccc',
+              can_pass_with_additional_verification: false,
+              attributes_requiring_additional_verification: [],
+              vendor_name: 'ResolutionMock',
+              vendor_workflow: nil,
+              verified_attributes: nil,
+            },
+            residential_address: {
+              success: true,
+              errors: {},
+              exception: nil,
+              timed_out: false,
+              transaction_id: '',
+              reference: '',
+              can_pass_with_additional_verification: false,
+              attributes_requiring_additional_verification: [],
+              vendor_name: 'ResidentialAddressNotRequired',
+              vendor_workflow: nil,
+              verified_attributes: nil,
+            },
+            state_id: {
+              success: true,
+              errors: {},
+              exception: nil,
+              mva_exception: nil,
+              requested_attributes: {},
+              timed_out: false,
+              transaction_id: 'state-id-mock-transaction-id-456',
+              vendor_name: 'StateIdMock',
+              verified_attributes: [],
+            },
+            threatmetrix: {
+              client: nil,
+              success: true,
+              errors: {},
+              exception: nil,
+              timed_out: false,
+              transaction_id: 'ddp-mock-transaction-id-123',
+              review_status: 'pass',
+              response_body: {
+                "fraudpoint.score": '500',
+                request_id: '1234',
+                request_result: 'success',
+                review_status: 'pass',
+                risk_rating: 'trusted',
+                summary_risk_score: '-6',
+                tmx_risk_rating: 'neutral',
+                tmx_summary_reason_code: ['Identity_Negative_History'],
+                first_name: '[redacted]',
+              },
+            },
+          },
+        },
+        ssn_is_unique: true,
+      },
+    )
+  end
+
+  let(:socure_idplus_base_url) { 'https://example.org' }
+
+  before do
+    document_capture_session.store_proofing_result(proofing_result.to_h)
+
+    allow(IdentityConfig.store).to receive(:socure_idplus_base_url).
+      and_return(socure_idplus_base_url)
+  end
+
+  describe '#perform' do
+    subject(:perform) do
+      allow(job).to receive(:create_analytics).and_return(analytics)
+
+      job.perform(
+        document_capture_session_result_id:,
+        encrypted_arguments:,
+        service_provider_issuer:,
+        user_email:,
+        user_uuid:,
+      )
+    end
+
+    let(:analytics) do
+      FakeAnalytics.new
+    end
+
+    let(:socure_response_body) do
+      {
+        referenceId: 'a1234b56-e789-0123-4fga-56b7c890d123',
```

Comment:
TIL that part of my subconscious brain is apparently dedicated to validating UUIDs because my brain immediately seized on there being a 'g' in here. Obviously this is not actually a problem in mock data. 😄

Reply:
99% sure I just copied this from the socure dev site

Reply:
@n1zyy Out of curiosity, why did you focus on there being a 'g' in the UUID?